ci/ci

```diff
@@ -38,27 +38,95 @@
 # subject to Altair's trademark licensing policies.
 
 import argparse
+import copy
 import fileinput
+import json
 import os
 import platform
 import shlex
+import shutil
 import subprocess
 import sys
+import threading
 import time
-
-pbs_dirname = ''
+from argparse import RawTextHelpFormatter
+from string import Template
+
+ci_dirname = ''
+# MACROS
+CONFIG_DIR = '.config_dir'
+STATUS_FILE = '.status'
+PARAM_FILE = '.params'
+REQUIREMENT_DECORATOR_FILE = '.requirements_decorator'
+CONFIGURE_OPT_FILE = '.configure_opt'
+BENCHPRESS_OPT_FILE = '.benchpress_opt'
+CONF_JSON_FILE = '.conf.json'
+
+requirements_template = Template('''num_servers=${num_servers}
+num_moms=${num_moms}
+num_comms=${num_comms}
+no_mom_on_server=${no_mom_on_server}
+no_comm_on_server=${no_comm_on_server}
+no_comm_on_mom=${no_comm_on_mom}
+''')
+
+service_template_prist = Template('''{
+"image": "${image}",
+"volumes": [
+    "../:/pbssrc",
+    "./:/src",
+    "./logs:/logs",
+    "./etc:/workspace/etc"
+],
+"entrypoint": "/workspace/etc/container-init",
+"environment": [
+    "NODE_TYPE=${node_type}",
+    "LANG=en_US.utf-8"
+],
+"networks": {
+    "ci.local": {
+    "aliases": [
+        "${alias}"
+        ]
+    }
+},
+"domainname": "ci.local",
+"container_name": "${hostname}",
+"hostname": "${hostname}",
+"user": "root",
+"privileged": true,
+"stdin_open": true,
+"tty": true
+}''')
 
 
 def log_error(str_to_print):
-    t = time.localtime()
-    current_time = time.strftime("%H:%M:%S", t)
-    print('\033[91m' + current_time + " ---> " + str_to_print + '\033[0m')
+    print("ERROR ::: " + str(str_to_print))
 
 
 def log_info(str_to_print):
     t = time.localtime()
     current_time = time.strftime("%H:%M:%S", t)
-    print('\033[1m' + current_time + " ---> " + str_to_print + '\033[0m')
+    print(current_time + " ---> " + str(str_to_print))
+
+
+def log_warning(str_to_print):
+    print("WARNING ::: " + str(str_to_print))
```

Comment:
why use single line functions?

Reply:
log_warning,log_error and log_info functions provide a standard way to communicate to the user on the command line, this follows with the design standard. Also if we add future changes (such as color encoding) then it will be easier to implement.